diff a/third_party/assimp/code/CMakeLists.txt b/third_party/assimp/code/CMakeLists.txt
--- a/third_party/assimp/code/CMakeLists.txt
+++ b/third_party/assimp/code/CMakeLists.txt
@@ -1176,13 +1176,6 @@ ADD_LIBRARY(assimp::assimp ALIAS assimp)
 
 TARGET_USE_COMMON_OUTPUT_DIRECTORY(assimp)
 
-# enable warnings as errors ########################################
-IF (MSVC)
-  TARGET_COMPILE_OPTIONS(assimp PRIVATE /WX)
-ELSE()
-  TARGET_COMPILE_OPTIONS(assimp PRIVATE -Werror)
-ENDIF()
-
 # adds C_FLAGS required to compile zip.c on old GCC 4.x compiler
 TARGET_COMPILE_FEATURES(assimp PRIVATE c_std_99)
